verification/policy.go

```diff
@@ -0,0 +1,153 @@
+	TrustStore string `json:"trustStore,omitempty"`
+	// TrustedIdentities this policy statement pins
+	TrustedIdentities []string `json:"trustedIdentities,omitempty"`
+}
+
+func isPresent(val string, values []string) bool {
+	for _, v := range values {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidatePolicyDocument validates a policy document according to it's version's rule set.
+// if any rule is violated, returns an error
+func ValidatePolicyDocument(policyDoc *PolicyDocument) error {
+	// Constants
+	wildcardScope := "*"
+	supportedPolicyVersions := []string{"1.0"}
+	supportedVerificationPresets := []string{"strict", "permissive", "audit", "skip"}
+	supportedTrustStorePrefixes := []string{"ca", "signingservice"}
+
+	// Validate Version
+	if !isPresent(policyDoc.Version, supportedPolicyVersions) {
+		return errors.New("Version '" + policyDoc.Version + "' is not supported")
+	}
+
+	// Validate the policy according to 1.0 rules
+	if len(policyDoc.TrustPolicies) == 0 {
+		return errors.New("Trust Policy document can not have zero statements")
+	}
+	var uniqueStatementNames []string
+	registryScopeCount := make(map[string]int)
+	for _, statement := range policyDoc.TrustPolicies {
+
+		// Verify statement name is valid
+		if statement.Name == "" {
+			return errors.New("Policy statement is missing a name")
+		} else {
+			if !isPresent(statement.Name, uniqueStatementNames) {
+				uniqueStatementNames = append(uniqueStatementNames, statement.Name)
+			}
+		}
+
+		// Verify registry scopes are valid
+		if len(statement.RegistryScopes) == 0 {
+			return errors.New("Policy statement has zero registry scopes")
+		} else {
+			if len(statement.RegistryScopes) > 1 && isPresent(wildcardScope, statement.RegistryScopes) {
+				return errors.New("Wildcard scope can not be shared with other registry scopes")
+			}
+			for _, scope := range statement.RegistryScopes {
+				registryScopeCount[scope]++
+			}
+		}
+
+		// Verify signature verification preset is valid
+		if !isPresent(statement.SignatureVerification, supportedVerificationPresets) {
+			return errors.New("SignatureVerification '" + statement.SignatureVerification + "' is not supported")
+		}
+
+		// strict and permissive verification needs trust store to verify authenticity
+		if statement.SignatureVerification == "strict" || statement.SignatureVerification == "permissive" {
+			if statement.TrustStore == "" {
+				return errors.New("Verification statement with strict or permissive preset is missing a trust store")
+			}
+		}
+
+		// Verify trust store type is valid
+		if statement.TrustStore != "" {
+			i := strings.Index(statement.TrustStore, ":")
+			if i < 0 || !isPresent(statement.TrustStore[:i], supportedTrustStorePrefixes) {
+				return errors.New("Statement '" + statement.Name + "' has a trust store with an unsupported trust store type")
+			}
+		}
+
+		// If there are trusted identitied, verify they are not empty
+		if len(statement.TrustedIdentities) > 0 {
```

Comment:
This if statement is not necessary, Go will not loop if `TrustedIdentities` is empty.